src/charts/grouped-bar.js

```diff
@@ -0,0 +1,988 @@
+                        }
+                    })
+                    return memo;
+                }, {});
+        }
+
+        /**
+         * @param  {HTMLElement} container DOM element that will work as the container of the graph
+         * @private
+         */
+        function buildSVG(container) {
+            if (!svg) {
+                svg = d3Selection.select(container)
+                    .append('svg')
+                    .classed('britechart grouped-bar', true);
+
+                buildContainerGroups();
+            }
+
+            svg
+                .attr('width', width)
+                .attr('height', height);
+        }
+
+        /**
+         * Parses dates and values into JS Date objects and numbers
+         * @param  {obj} data Raw data from JSON file
+         * @return {obj}      Parsed data with values and dates
+         */
+        function cleanData(data) {
+            return data.map((d) => {
+                d.value = +d[valueLabel];
+                d.group = d[groupLabel];
+                // for tooltip
+                d.topicName = getGroup(d);
+                d.name = d[nameLabel];
+
+                return d;
+            });
+        }
+
+        /**
+         * Draws the x and y axis on the svg object within their
+         * respective groups
+         * @private
+         */
+        function drawAxis() {
+            if (!horizontal) {
+                svg.select('.x-axis-group .axis.x')
+                    .attr('transform', `translate( 0, ${chartHeight} )`)
+                    .call(xAxis);
+
+                svg.select('.y-axis-group.axis')
+                    .attr('transform', `translate( ${-xAxisPadding.left}, 0)`)
+                    .call(yAxis)
+                    .call(adjustYTickLabels);
+            } else {
+                svg.select('.x-axis-group .axis.x')
+                    .attr('transform', `translate( 0, ${chartHeight} )`)
+                    .call(xAxis);
+
+                svg.select('.y-axis-group.axis')
+                    .attr('transform', `translate( ${-xAxisPadding.left}, 0)`)
+                    .call(yAxis);
+            }
+        }
+
+        /**
+         * Adjusts the position of the y axis' ticks
+         * @param  {D3Selection} selection Y axis group
+         * @return void
+         */
+        function adjustYTickLabels(selection) {
+            selection.selectAll('.tick text')
+                .attr('transform', `translate(${yTickTextXOffset}, ${yTickTextYOffset})`);
+        }
+
+        /**
+         * Draws grid lines on the background of the chart
+         * @return void
+         */
+        function drawGridLines(xTicks, yTicks) {
+            if (grid === 'horizontal' || grid === 'full') {
+                svg.select('.grid-lines-group')
+                    .selectAll('line.horizontal-grid-line')
+                    .data(yScale.ticks(yTicks).slice(1))
+                    .enter()
+                    .append('line')
```

Comment:
Here we want the 'append' to have 2 spaces of indentation, to mark where we are changing the selection